include/sdf/Error.hh

```diff
@@ -159,20 +159,31 @@ namespace sdf
     /// \brief Constructor.
     /// \param[in] _code The error code.
     /// \param[in] _message A description of the error.
+    /// \param[in] _xmlPath The XPath-like trace that is related to this error.
+    /// \sa ErrorCode.
+    public: Error(const ErrorCode _code, const std::string &_message,
+                  const std::string &_xmlPath);
+
+    /// \brief Constructor.
+    /// \param[in] _code The error code.
+    /// \param[in] _message A description of the error.
+    /// \param[in] _xmlPath The XPath-like trace that is related to this error.
     /// \param[in] _filePath The file path that is related to this error.
     /// \sa ErrorCode.
     public: Error(const ErrorCode _code, const std::string &_message,
-                  const std::string &_filePath);
+                  const std::string &_xmlPath, const std::string &_filePath);
```

Comment:
We'd probably like to have this feature in `sdf11`, so that would be a suitable patch that doesn't break ABI for now.

So far, I've found that these fields in `Error` do not exist independently from each other, where if the `Error` has a line number, it will definitely have a path and an XML path, whereas having a path, guarantees an XML path, but not necessarily a line number.

I'd be more inclined to remove all the overloads added in https://github.com/osrf/sdformat/pull/512, and just use setters instead in the next release. :thinking:

Reply:
> I'd be more inclined to remove all the overloads added in #512, and just use setters instead in the next release. thinking

Sounds good, but they have to first be deprecated before being removed.

Reply:
Awesome, I'll ask you more about that work flow during our weekly sync. For now I'll go ahead with the setters.

Reply:
https://github.com/osrf/sdformat/pull/548/commits/ece5cb0fe3f4682af695f88fb99046b22b721745

Reply:
@jennuine FYI